browser/resources/settings/brave_search_engines_page/brave_search_engines_page.ts

```diff
@@ -68,12 +71,19 @@ class BraveSearchEnginesPage extends BraveSearchEnginesPageBase {
       'private-search-engines-changed', updatePrivateSearchEngines)
   }
 
+  override currentRouteChanged() {
+    this.showPrivateSearchEngineListDialog_ =
+      Router.getInstance().getCurrentRoute() === routes.PRIVATE_SEARCH
+  }
+
   private onOpenPrivateDialogButtonClick_() {
     this.showPrivateSearchEngineListDialog_ = true
```

Comment:
nit: maybe this assignment is not needed anymore as it's updated by `currentRouteChanged()`?

Reply:
Good point, removed!